src/Models/BaseElement.php

```diff
@@ -528,6 +536,85 @@ public function getContentForSearchIndex(): string
         return $content;
     }
 
+    /**
+     * Provides content for CMS search
+     */
+    public function getContentForCmsSearch(): string
+    {
+        $textualDatabaseFields = $this->getTextualDatabaseFields();
+        $excludedFields = $this->getFieldsExcludedFromCmsSearch();
+        $contents = [];
+        foreach ($this->config()->get('db') as $fieldName => $fieldType) {
+            if (!in_array($fieldType, $textualDatabaseFields) || in_array($fieldName, $excludedFields)) {
+                continue;
+            }
+            $contents[] = $this->$fieldName;
+        }
+        // Use |#| to delimit different fields rather than space so that you don't
+        // accidentally join results of two columns that are next to each other in a table
+        $delimiter = '|#|';
+        $content = implode($delimiter, $contents);
+
+        // Strips tags and be sure there's a space between words.
+        $content = trim(strip_tags(str_replace('<', ' <', $content)));
+
+        // Allow projects to update content of third-party elements.
+        $content .= $delimiter;
+        $this->extend('updateContentForCmsSearch', $content);
+
+        // Remove delimiters that aren't required
+        $content = $this->removeExtraDelimiters($content, $delimiter);
+
+        return $content;
+    }
+    /**
+     * Get fields that have a Varchar or Text like type in the database
+     */
+    private function getTextualDatabaseFields(): array
+    {
+        $textualDatabaseFields = singleton(DataObjectSchema::class)->databaseFields($this);
```

Comment:
```suggestion
        $textualDatabaseFields = DataObject::getSchema()->databaseFields($this);
```
This is the standard way to get that singleton

Reply:
Updated